portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,53 +19,73 @@
 import com.google.appengine.api.datastore.Entity;
 import com.google.appengine.api.datastore.PreparedQuery;
 import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 import com.google.gson.Gson;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.util.ArrayList;
+import java.util.List;
+import java.util.Collections;
 
 
 /**
-  * Servlet that returns some example content. 
+  * Servlet that returns comments in database. 
   * Note: We first added the Gson library dependency to pom.xml.
   */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
+  int maxNumComments = 1;
+  ArrayList<String> msglist = new ArrayList<String>();
   String comment = "Comment";
+  String contentProperty = "content";
+  String timestampProperty = "timestamp";
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String inputNumComment = "numComment";
+    maxNumComments = Integer.parseInt(request.getParameter(inputNumComment));
+    
     // Create the query and prepared query to load comment entities from database
-    Query query = new Query(comment);
+    Query query = new Query(comment).addSort(timestampProperty, SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
     
-    // Add all comment contents to the msgList  
+    // Add all comment contents to the msglist  
     ArrayList<String> msglist = new ArrayList<String>();
     for (Entity comment:results.asIterable()) {
-      String commentMsg = (String)comment.getProperty("content");
+      String commentMsg = (String)comment.getProperty(contentProperty);
       msglist.add(commentMsg);
     }
+    Collections.shuffle(msglist);
+
+    // Limit number of comments;
+    List<String> updatedStrList;
```

Comment:
Question: Why use List instead of ArrayList here?

Reply:
ArrayList.sublist() method returns back a List object instead of an ArrayList object. If I don't use List for updatedStrList, I have to do a downcasting to change it back to ArrayList.

Reply:
Updated: No need for this based on change of method